modules/standard/Reflection.chpl

```diff
@@ -50,18 +50,46 @@ private proc checkQueryT(type t) type {
 proc numFields(type t) param : int
   return __primitive("num fields", checkQueryT(t));
 
+/* Get the name of the ith field in a class or record.
+   Causes a compilation error if `idx` is not in 0..<numFields(t).
+
+   :arg t: a class or record type
+   :arg idx: which field to get the name of
+   :returns: the name of the field, as a param string
+ */
+proc getFieldName(type t, param idx:int) param : string
+  return __primitive("field num to name", checkQueryT(t), idx+1);
+
 /* Get the name of the ith field in a class or record.
    Causes a compilation error if `i` is not in 0..<numFields(t).
 
    :arg t: a class or record type
    :arg i: which field to get the name of
    :returns: the name of the field, as a param string
  */
-proc getFieldName(type t, param i:int) param : string
+pragma "last resort"
+deprecated "Formal 'i' is deprecated, please use 'idx' instead"
+proc getFieldName(type t, param i:int) param : string {
   return __primitive("field num to name", checkQueryT(t), i+1);
```

Comment:
Might be good to call the new function in the old function, in case someone tries to make a code change and fails to notice which one they are modifying.